rust/lance/src/index/vector/pq.rs

```diff
@@ -277,10 +332,28 @@ impl VectorIndex for PQIndex {
 
         debug_assert_eq!(distances.len(), row_ids.len());
 
+        // Remove any tombstone rows from consideration when sorting
+        let distance_ids = distances
+            .as_any()
+            .downcast_ref::<Float32Array>()
+            .unwrap()
+            .values()
+            .iter()
+            .copied()
+            .zip(row_ids.values().iter().copied())
+            .filter(|(_, row_id)| *row_id != RowId::TOMBSTONE_ROW)
```

Comment:
Is this before remapping is done or after?

If the remapping is done, why there still are tomestone rows in the index file?